### DISCUSSION TITLE
Running the library on a custom dataset

### DISCUSSION THREAD

**Discussion by enislalmi** [open-edge-platform/anomalib#1046] [Q&A]
I am trying to run the library on a custom dataset. I have created the config file, following [this](https://github.com/openvinotoolkit/anomalib/discussions/751). However when I create `datamodule = get_datamodule(config)`, I get this error `ConfigAttributeError: Missing key mask_dir
    full_key: dataset.mask_dir
    object_type=dict`, even though I have specified the task is classification, and the mask is optional. Any resolving ideas? Thanks!

**Comment by fibr1900** [comment]
I had the same problem, but it turned out that the problem was that I was using an "old" version of Anomalib when I followed the "new" config structures. For example, the new mask parameter name is "mask_dir", while the old version was only "mask". You might just need to make sure you are using the right version of Anomalib.
